File: count-examples/count.rb
file_path = ARGV[0] || "input.txt"
WORDS = {}
file = File.open(file_path, "r") 
 
file.each_line do |line|
  words = line.split 
  words.each do |word|
    word = word.gsub(/[,()'.;?!"]/,'').downcase
    if WORDS[word]
      WORDS[word] += 1
    else
      WORDS[word] = 1
    end
  end
end
 
WORDS.sort {|a,b| a[1] <=> b[1]}.each do |key,value|
  puts "#{key} => #{value}"
end

File: count-examples/count.py
'''
Count the number of unique words from the sample input file
ignoring special characters and print the results ordered by count.
'''

import re

with open('input.txt') as f:
    lines = f.read()

words = str.split(lines.lower())
counts = dict()

for word in words:
    cleanWord = re.sub(r'\W+','', word)
    if cleanWord in counts:
        counts[cleanWord] += 1
    else:
        counts[cleanWord] = 1

for k, v in sorted(counts.items(), key=lambda item: item[1]):
    print(f'{k:6s} => {v}')
